src/soko_loader.rs

```diff
@@ -158,3 +178,20 @@ pub fn load_level(contents: &str) -> Result<World, String> {
         _ => Err("Level must start with a title".to_string()),
     }
 }
+
+pub fn parse_sokoban_worlds(sokoban_text: &str) -> Result<Vec<World>, String> {
+    let tokens = tokenize(sokoban_text);
+    if tokens.is_none() {
+        return Err("Level failed to load".to_string());
+    }
+    let tokens = tokens.unwrap();
+
+    let sokoban_token_groups = group_sokoban_tokens(tokens);
+    let mut worlds = Vec::new();
+    for level in sokoban_token_groups {
```

Comment:
ya I think this could be smthn like:
```
worlds = sokoban_token_groups
    .iter()
    .map(|level| parse_sokoban_level(level.clone()))
    .filter(|level| level.is_some())
    .map(|level| level.unwrap()) // safe because we filtered out the Nones
    .collect()
```

Reply:
looks nicer than my code but im failing to get it to work